aten/src/ATen/native/SummaryOps.cpp

```diff
@@ -52,10 +51,77 @@ Tensor _bincount_cpu_template(
   }
   return output;
 }
+
+// A helper function to get the bin for histogram
+// while each bin is inclusive at the lower end and exclusive at the higher,
+// i.e. [start, end) the last bin is inclusive at both, i.e. [start, end], in
+// order to include maxvalue if exists
+template <typename input_t>
+inline int64_t getbin(input_t x, input_t min, input_t max, int64_t bins) {
+  if (x == max)
+    return bins - 1;
+  return (int64_t)((x - min) * bins / (max - min));
+}
+
+///////////////// histogram /////////////////
+template <typename input_t, typename weights_t>
+Tensor _histogram_cpu_template_uniformbins(
+    const Tensor& self,
+    int64_t nbins,
+    const Tensor& weights,
+    input_t min,
+    input_t max,
+    bool density) {
+
+  if (min == max) {
+    min -= 1;
+    max += 1;
+  }
+  TORCH_CHECK(nbins > 0, "bins must be > 0");
+  // Check if range is finite, hack to keep MSVC from complaining
+  TORCH_CHECK(
```

Comment:
nit: you could avoid this check for non-floating point types and not have to cast